packages/cache/__tests__/hydration.js

```diff
@@ -35,3 +35,145 @@ test('rehydrated styles to head can be flushed', () => {
+  const css = `color:hotpink;`
+  const hash = hashString(css)
+  safeQuerySelector('head').innerHTML = [
+    '<style data-emotion="emo 1lrxbo5">.emo-1lrxbo5{color:hotpink;}</style>',
+    '<style data-emotion="css qweqwee">.css-qweqwee{color:red;}</style>'
+  ].join('')
+
+  // this moves emotion style tags at initialization time
+  jest.resetModules()
+  require('@emotion/react')
+
+  const cache = createCache({ key: 'emo' })
+  expect(document.documentElement).toMatchInlineSnapshot(`
+    <html>
+      <head>
+        <style
+          data-emotion="css qweqwee"
+          data-s=""
+        >
+          .css-qweqwee{color:red;}
+        </style>
+        <style
+          data-emotion="emo 1lrxbo5"
+          data-s=""
+        >
+          .emo-1lrxbo5{color:hotpink;}
+        </style>
+      </head>
+      <body />
+    </html>
+  `)
+
+  cache.sheet.flush()
+  expect(document.documentElement).toMatchInlineSnapshot(`
+    <html>
+      <head>
+        <style
+          data-emotion="css qweqwee"
+          data-s=""
+        >
+          .css-qweqwee{color:red;}
+        </style>
+      </head>
+      <body />
+    </html>
+  `)
+})
+
+test('Should only hydrate style elements matching the cache key', () => {
+  let css = `color:hotpink;`
+  let hash = hashString(css)
+
+  safeQuerySelector(
+    'body'
+  ).innerHTML = `<style data-emotion="emo ${hash}">.css-${hash}{${css}}</style>`
+
+  const cache = createCache({ key: 'css' })
+
+  expect(cache.inserted).toEqual({})
+  expect(document.documentElement).toMatchInlineSnapshot(`
+    <html>
+      <head>
+        <style
+          data-emotion="emo 1lrxbo5"
+          data-s=""
+        >
+          .css-1lrxbo5{color:hotpink;}
+        </style>
+      </head>
+      <body />
+    </html>
+  `)
+
+  const cache2 = createCache({ key: 'emo' })
+
+  expect(cache2.inserted).toEqual({ [hash]: true })
+  expect(document.documentElement).toMatchInlineSnapshot(`
+    <html>
+      <head>
+        <style
+          data-emotion="emo 1lrxbo5"
+          data-s=""
+        >
+          .css-1lrxbo5{color:hotpink;}
+        </style>
+      </head>
+      <body />
+    </html>
+  `)
+})
+
+describe('In production mode', () => {
+  beforeEach(() => {
+    process.env.NODE_ENV = 'production'
+  })
+
+  afterEach(() => {
+    process.env.NODE_ENV = 'test'
+  })
+
+  test('Existing head styles from Emotion 10 should not be moved', () => {
+    safeQuerySelector('head').innerHTML = [
+      '<style data-emotion="css-global"></style>',
+      '<style data-emotion="css"></style>',
+      '<style>.my-div{color: red;}</style>'
+    ].join('')
+
+    const css = `color:hotpink;`
+    const hash = hashString(css)
+
+    safeQuerySelector(
+      'body'
+    ).innerHTML = `<style data-emotion="css ${hash}">.css-${hash}{${css}}</style>`
+
+    createCache({ key: 'css' })
+
+    expect(document.documentElement).toMatchInlineSnapshot(`
```

Comment:
based on this assertion it's rather hard to see if this asserts the correct thing - maybe it would be better to render those Emotion 10 styles to the body? that way we'd know that they have not been moved to the head

Reply:
Yea this test is a bit of a weird one tbh. It's intended to assert the cause of our original issue which is essentially style tags on the head being re-mounted to the head when a cache provider instantiates itself, resulting in CSSOM styles being lost. 

Since we're not able to test CSSOM styles in jest, I feel like it's probably a good idea just to remove it instead. 
What do you think?  😄

Reply:
This is the only test that tests behavior that has changed, I've updated it so it's more clear